service/src/main/java/com/codeforcommunity/processor/ProcessorImpl.java

```diff
@@ -46,14 +46,16 @@ public ProcessorImpl(DSLContext db) {
   }
 
   @Override
-  public boolean createNews(String title, String description, String author, LocalDateTime date, String content) {
+  public Optional<NewsReturn> createNews(String title, String description, String imageUrl, String author, LocalDateTime date, String content) {
     try {
-      db.insertInto(Tables.NEWS, Tables.NEWS.TITLE, Tables.NEWS.DESCRIPTION, Tables.NEWS.AUTHOR, Tables.NEWS.DATE,
-          Tables.NEWS.CONTENT).values(title, description, author, Timestamp.valueOf(date), content).execute();
+      Result result = db.insertInto(Tables.NEWS, Tables.NEWS.TITLE, Tables.NEWS.DESCRIPTION, Tables.NEWS.IMAGE_URL, Tables.NEWS.AUTHOR, Tables.NEWS.DATE,
+              Tables.NEWS.CONTENT).values(title, description, imageUrl, author, Timestamp.valueOf(date), content)
+              .returning(Tables.NEWS.ID)
+              .fetch();
+      return getNews((int) result.getValue(0,0));
```

Comment:
This block of logic can be made a lot clearer by utilizing a newsRecord object that jOOQ provides. Check in `persist/target/generated-sources/jooq/...` for details or ask Ryan Jung.